dsgrid/helpers.py

```diff
@@ -31,17 +34,49 @@ def ensure_enum(cls, val):
         return cls[val]
     return cls(val)
 
+
 def lighten_color(hex_color,fraction_to_white):
     rgb_color = np.array(webcolors.hex_to_rgb(hex_color))
     white = np.array([255,255,255])
     direction = white - rgb_color
     result = [int(round(x)) for x in list(rgb_color + direction * fraction_to_white)]
     return webcolors.rgb_to_hex(tuple(result))
     
+
 def palette(hex_color,n,max_fraction=0.75):
     result = []; step = max_fraction / float(n)
     for frac in [i * step for i in range(n)]:
         result.append(lighten_color(hex_color,frac))
     assert len(result) == n
     return result
 
+
+class h5Reader(object):
+    def __init__(self, filepath):
+        self.filepath = filepath
+        if self.is_hsds:
+            import h5pyd
+            self._f = h5pyd.File(filepath, mode="r", use_cache=False)
+        elif self.is_s3:
+            import s3fs
+            self._s3p = s3fs.S3FileSystem().open(filepath, 'rb')
+            self._f = h5py.File(self._s3p, mode="r")
+        else:
+            self._f = h5py.File(filepath, mode="r")
+
+    @property
+    def is_hsds(self):
+        return str(self.filepath).startswith("/nrel/")
+
+    @property
+    def is_s3(self):
+        return str(self.filepath).startswith("s3://")
+
+    def __enter__(self):
+        return self._f.__enter__()
```

Comment:
This is probably a matter of style but I probably would have returned `self._f` here and then called `self._f.close()` in `__exit__`.

Reply:
I concur with @daniel-thom , see this example in rex:
https://github.com/NREL/rex/blob/2dc74a77c37eb00bccf391f24cc9aa2a0b53fbe9/rex/resource.py#L568-L597